From f26295a7312c7a2859d360375580b45b11625b9e Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Wed, 6 Mar 2013 08:18:25 +0100
Subject: [PATCH 11/15] Don't strip rc. from sysv init scripts

This was added for Frugalware but in our case strips of rc. from
rc.local which we don't want.
---
 src/core/service.c |   22 ++--------------------
 1 file changed, 2 insertions(+), 20 deletions(-)

diff --git a/src/core/service.c b/src/core/service.c
index 2d322bf..e5313a1 100644
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -336,9 +336,6 @@ static char *sysv_translate_name(const char *name) {
         if (endswith(name, ".sh"))
                 /* Drop .sh suffix */
                 strcpy(stpcpy(r, name) - 3, ".service");
-        else if (startswith(name, "rc."))
-                /* Drop rc. prefix */
-                strcpy(stpcpy(r, name + 3), ".service");
         else
                 /* Normal init script name */
                 strcpy(stpcpy(r, name), ".service");
@@ -999,10 +996,8 @@ static int service_load_sysv_name(Service *s, const char *name) {
         assert(s);
         assert(name);
 
-        /* For SysV services we strip the rc.* and *.sh
-         * prefixes/suffixes. */
-        if (startswith(name, "rc.") ||
-            endswith(name, ".sh.service"))
+        /* For SysV services we strip the *.sh suffixes. */
+        if (endswith(name, ".sh.service"))
                 return -ENOENT;
 
         STRV_FOREACH(p, UNIT(s)->manager->lookup_paths.sysvinit_path) {
@@ -1025,19 +1020,6 @@ static int service_load_sysv_name(Service *s, const char *name) {
                 }
                 free(path);
 
-                if (r >= 0 && UNIT(s)->load_state == UNIT_STUB) {
-                        /* Try rc.* init scripts */
-
-                        path = strjoin(*p, "/rc.", name, NULL);
-                        if (!path)
-                                return -ENOMEM;
-
-                        /* Drop .service suffix */
-                        path[strlen(path)-8] = 0;
-                        r = service_load_sysv_path(s, path);
-                        free(path);
-                }
-
                 if (r < 0)
                         return r;
 
-- 
1.7.10.4

